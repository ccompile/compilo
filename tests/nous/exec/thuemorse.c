
/* Affichage des premiers termes de la suite de Thue-Morse */

int compter_uns(int x)
{
    int res;

    res = 0;
    while(x != 0)
    {
        res = (res + (x % 2)) % 2;
        x = x / 2;
    }
    
    return res;
}

void thuemorse(int n)
{
    int i;

    for(i = 0; i < n; i++)
        putchar('0' + compter_uns(i));
    putchar(10);
}

int main()
{
    thuemorse(128);
    return 0;
}



