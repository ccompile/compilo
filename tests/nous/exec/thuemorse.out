01101001100101101001011001101001100101100110100101101001100101101001011001101001011010011001011001101001100101101001011001101001
